### PR TITLE
CI: use mdformat

### PR DESCRIPTION
**Pull request by Borda**
# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## What does this PR do?
fixed by https://github.com/executablebooks/mdformat/issues/246

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -16,27 +16,29 @@ We are always looking for help implementing new features or fixing bugs.
    - Add details on how to reproduce the issue - a minimal test case is always best, colab is also great.
      Note, that the sample code shall be minimal and if needed with publicly available data.
 
-2. Try to fix it or recommend a solution. We highly recommend to use test-driven approach:
+1. Try to fix it or recommend a solution. We highly recommend to use test-driven approach:
```
Comment (ethanwharris) [inline .github/CONTRIBUTING.md:19]: Looks like this should stay as 2
```suggestion
2. Try to fix it or recommend a solution. We highly recommend to use test-driven approach:
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -16,27 +16,29 @@ We are always looking for help implementing new features or fixing bugs.
...
 
    - Convert your minimal code example to a unit/integration test with assert on expected results.
    - Start by debugging the issue... You can run just this particular test in your IDE and draft a fix.
    - Verify that your test case fails on the master branch and only passes with the fix applied.
 
-3. Submit a PR!
+1. Submit a PR!
```
Comment (ethanwharris) [inline .github/CONTRIBUTING.md:25]: And this should be 3
```suggestion
3. Submit a PR!
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -16,27 +16,29 @@ We are always looking for help implementing new features or fixing bugs.
...
 
 ### New Features:
 
 1. Submit a github issue - describe what is the motivation of such feature (adding the use case or an example is helpful).
-2. Let's discuss to determine the feature scope.
-3. Submit a PR! We recommend test driven approach to adding new features as well:
+
+1. Let's discuss to determine the feature scope.
+
+1. Submit a PR! We recommend test driven approach to adding new features as well:
```
Comment (ethanwharris) [inline .github/CONTRIBUTING.md:36]: Lot's of erros with numberred lists haha

**Review by ethanwharris — COMMENTED** [review]
It really doesn't like numberred bullets haha

**Comment by Borda** [comment]
> It really doesn't like numberred bullets haha

MD does not require any number, so in fact, if you write everywhere 9, you'll get the indexing from 1
see https://guides.github.com/features/mastering-markdown/

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (Borda) [reply .github/CONTRIBUTING.md:36]: there are no errors :D

**Review by ethanwharris — APPROVED** [review]

**Pull request merged**